compatibility_spark-3/src/test/scala/za/co/absa/hyperdrive/compatibility/impl/writer/delta/TestDeltaUtil.scala

```diff
@@ -0,0 +1,102 @@
+  behavior of "DeltaUtil"
+
+  "createDeltaTableIfNotExists" should "create delta table if destination directory is empty" in {
+    val schema = StructType(Seq(StructField("testColumn", StringType, nullable = false)))
+
+    DeltaUtil.createDeltaTableIfNotExists(spark, baseDirPath, schema, Seq())
+
+    FileUtils.readFileLines(s"$baseDirPath/_delta_log/00000000000000000000.json").nonEmpty should be(true)
+  }
+
+  it should "create delta table if destination directory does not exist" in {
+    val schema = StructType(Seq(StructField("testColumn", StringType, nullable = false)))
+
+    DeltaUtil.createDeltaTableIfNotExists(spark, destinationPath, schema, Seq())
+
+    FileUtils.readFileLines(s"$destinationPath/_delta_log/00000000000000000000.json").nonEmpty should be(true)
+  }
+
+  it should "throw exception if destination directory is not empty" in {
+    val schema = StructType(Seq(StructField("testColumn", StringType, nullable = false)))
+
+    new File(baseDirPath + "/filename.txt").createNewFile()
+
+    assertThrows[IllegalArgumentException](DeltaUtil.createDeltaTableIfNotExists(spark, baseDirPath, schema, Seq()))
+  }
+
+  "getDataFrameWithSortColumns" should "return dataframe with sort columns" in {
+    import spark.implicits._
+    val timestamp = Timestamp.valueOf("2022-11-02 14:15:39.11973")
+
+    val testInput: DataFrame = Seq(
+      CDCEvent("1", "value1", timestamp, "PT"),
+      CDCEvent("2", "value2", timestamp, "UP"),
+      CDCEvent("3", "value3", timestamp, "DL"),
+      CDCEvent("4", "value4", timestamp, "XX")
+    ).toDF
+
+    val expected: Seq[Row] = Seq(
+      Row("1", "value1", timestamp, "PT", 1, "value1"),
+      Row("2", "value2", timestamp, "UP", 2, "value2"),
+      Row("3", "value3", timestamp, "DL", 3, "value3"),
+      Row("4", "value4", timestamp, "XX", 0, "value4")
+    )
+
+    val sortFieldsPrefix = "_tmp_hyperdrive_"
+    val precombineColumns = Seq(
+      "eventType",
+      "value"
+    )
+    val precombineColumnsCustomOrder = Map("eventType" -> Seq("PT", "UP", "DL"))
+
+    val result = DeltaUtil.getDataFrameWithSortColumns(
+      testInput,
+      sortFieldsPrefix,
+      precombineColumns,
+      precombineColumnsCustomOrder
+    )
+
+    result.collect().toSeq should contain theSameElementsAs expected
+    result.schema.fieldNames should contain theSameElementsAs precombineColumns.map(col => s"$sortFieldsPrefix$col") ++ testInput.schema.fieldNames
+  }
+
+  "isDirEmptyOrDoesNotExist" should "return true if directory is empty" in {
+    DeltaUtil.isDirEmptyOrDoesNotExist(spark, baseDirPath) should be(true)
+  }
+
+  it should "return true if directory does not exist" in {
+    DeltaUtil.isDirEmptyOrDoesNotExist(spark, destinationPath) should be(true)
+  }
+
+  it should "return false if directory contains a file" in {
+    new File(baseDirPath + "/filename.txt").createNewFile()
+
+    DeltaUtil.isDirEmptyOrDoesNotExist(spark, baseDirPath) should be(false)
+  }
```

Comment:
Add the following case:
- Path exists, but is not a directory

Reply:
Done